### PR TITLE
Remove RestrictedRegister fields and functions from OMR::Z::CodeGenerator

### PR DESCRIPTION
**Pull request by nbhuiyan**
This pull request is for the removal of the following fields and functions:
```plaintext
setCurrentlyClobberedRestrictedRegister
clearCurrentlyClobberedRestrictedRegister
getCurrentlyClobberedRestrictedRegister
setKilledRestrictedRegisters
getKilledRestrictedRegisters
setCurrentlyRestrictedRegisters
getCurrentlyRestrictedRegisters
```

These fields and functions are not used in OMR or any other downstream projects.

Issue: #2063
Signed-off-by: Nazim Uddin Bhuiyan <nazimudd@ualberta.ca>

### CODE REVIEW

**Comment by 0xdaryl** [comment]
Is this a WIP because there are more commits coming, or because you're testing the one commit that's here?  Not sure if you accidentally forgot to push something...   :-)

**Comment by nbhuiyan** [comment]
There are more commits coming. I had to leave this to work on something else in openj9. Should be done soon!

**Comment by nbhuiyan** [comment]
@0xdaryl This PR is ready for review.

File: compiler/z/codegen/OMRCodeGenerator.cpp [hunk 1]
```diff
@@ -10281,43 +10273,8 @@ OMR::Z::CodeGenerator::setUnavailableRegisters(TR::Block *b, TR_BitVector &unava
 int8_t
 OMR::Z::CodeGenerator::getCurrentRegisterPressure(TR_RegisterKinds rk)
```
Comment (0xdaryl) [inline compiler/z/codegen/OMRCodeGenerator.cpp]: Please remove this function as well and keep folding code that calls it assuming it always returns `0`.  It seems to have been added specifically to support restricted registers.  There are a few places in the Z evaluators that call this, so there are some folding opportunities there.

**Comment by 0xdaryl** [comment]
@genie-omr build all

**Pull request merged**